src/store/app.ts

```diff
@@ -27,30 +39,91 @@ export const useAppStore = defineStore('app', {
 
             try {
-                await keycloak.init({
+                const authenticated = await keycloak.init({
                     onLoad: 'check-sso',
                     silentCheckSsoRedirectUri: silentCheckSsoHtmlUrl,
                     redirectUri: '/',
                 })
+
                 this.keycloak = keycloak
-                this.isLoggedIn = keycloak.authenticated ?? false
-                if (this.isLoggedIn) {
-                    const currentUser = await useClient().getCurrentUser()
-                    this.currentUserId = currentUser.currentUser?.id
+                this.isLoggedIn = authenticated
+
+                if (logToken === true) {
+                    console.log('Token', keycloak.token)
                 }
-                console.log('Token', keycloak.token)
             } catch (error) {
                 console.error('Failed to initialize adapter:', error)
             }
+
+            if (this.isLoggedIn) {
+                try {
+                    const currentUser = await useClient().getCurrentUser()
+                    this.currentUserId = currentUser.currentUser?.id
+                } catch (error) {
+                    console.error(
+                        'Failed to get the user and determine their user ID:',
+                        error
+                    )
+                }
+
+                try {
+                    const keycloakRealmAccess = keycloak.realmAccess
+                    if (
+                        keycloakRealmAccess === null ||
+                        keycloakRealmAccess === undefined
+                    ) {
+                        throw new Error(
+                            'The property realmAccess of the initialized Keycloak instance is either null or undefined.'
+                        )
+                    } else {
+                        const roles = keycloak.realmAccess?.roles
+                        if (roles === undefined) {
+                            return
+                        } else {
+                            const additionalUserRoles: UserRole[] = []
+                            if (
+                                roles.includes(UserRole.Employee.toLowerCase())
+                            ) {
+                                additionalUserRoles.push(UserRole.Employee)
+                                this.activeUserRole = UserRole.Employee
+                            }
+                            if (
+                                roles.includes(UserRole.SiteAdmin.toLowerCase())
+                            ) {
+                                additionalUserRoles.push(UserRole.SiteAdmin)
+                                this.activeUserRole = UserRole.SiteAdmin
+                            }
+
+                            this.userRolesOfCurrentUser.push(
+                                ...additionalUserRoles
+                            )
+
+                            console.log(this.userRolesOfCurrentUser)
```

Comment:
this logs a ref which contains a list
the output is completely unreadable
see https://vuejs.org/api/reactivity-advanced.html#toraw